tests/batch_2/SingleOwnerTonWallet.admin.spec.ts

```diff
@@ -0,0 +1,344 @@
+                {
+                    owner_address: owner.address,
+                    admin_address: admin.address,
+                },
+                sotwCode,
+            ),
+        );
+
+        deployer = await blockchain.treasury('deployer');
+
+        const deployResult = await singleOwnerTonWallet.sendDeploy(deployer.getSender(), toNano('0.05'));
+
+        expect(deployResult.transactions).toHaveTransaction({
+            from: deployer.address,
+            to: singleOwnerTonWallet.address,
+            deploy: true,
+            success: true,
+        });
+    });
+
+    it(`can transfer admin`, async () => {
+        const originalAdminAddress = (await singleOwnerTonWallet.getStorage())?.admin_address;
+        expect(originalAdminAddress?.equals(admin.address)).toEqual(true);
+        const newAdmin = await blockchain.treasury('newAdmin');
+        const totalSent = toNano('0.5');
+        const sendTransferAdminResult = await singleOwnerTonWallet.sendTransferAdmin(
+            admin.getSender(),
+            totalSent,
+            newAdmin.address,
+        );
+        const updatedAdminAddress = (await singleOwnerTonWallet.getStorage())?.admin_address;
+
+        expect(sendTransferAdminResult.transactions).toHaveTransaction({
+            from: admin.address,
+            to: singleOwnerTonWallet.address,
+            success: true,
+            op: Number(SingleOwnerTonWalletOpcode.transfer_admin),
+        });
+        expect(updatedAdminAddress?.equals(newAdmin.address)).toEqual(true);
+    });
+
+    it(`should refund excess toncoin on transfer admin is success`, async () => {
+        const newAdmin = await blockchain.treasury('newAdmin');
+        const totalSent = toNano('0.5');
+        const sendTransferAdminResult = await singleOwnerTonWallet.sendTransferAdmin(
+            admin.getSender(),
+            totalSent,
+            newAdmin.address,
+        );
+        const updatedAdminAddress = (await singleOwnerTonWallet.getStorage())?.admin_address;
+
+        expect(sendTransferAdminResult.transactions).toHaveTransaction({
+            from: admin.address,
+            to: singleOwnerTonWallet.address,
+            success: true,
+            op: Number(SingleOwnerTonWalletOpcode.transfer_admin),
+        });
+        expect(updatedAdminAddress?.equals(newAdmin.address)).toEqual(true);
+        // excess
+        expect(sendTransferAdminResult.transactions).toHaveTransaction({
+            from: singleOwnerTonWallet.address,
+            to: admin.address,
+            success: true,
+            op: Number(SingleOwnerTonWalletOpcode.excesses),
+        });
+
+        const refundTx = flattenTransaction(
+            findTransactionRequired(sendTransferAdminResult.transactions, {
+                from: singleOwnerTonWallet.address,
+                to: admin.address,
+                success: true,
+                op: Number(SingleOwnerTonWalletOpcode.excesses),
+            }),
+        );
```

Comment:
```suggestion
        const refundTx = flattenTransaction(
            findTransactionRequired(sendTransferAdminResult.transactions, {
                from: singleOwnerTonWallet.address,
                to: admin.address,
                success: true,
                op: Number(SingleOwnerTonWalletOpcode.excesses),
            }),
        );
```

findTransactionRequired will throw if tx not found already

Reply:
60a8b84c14daaa56bc610fdc432dbe8813d3cda6